client/my-sites/plans-features-main/index.tsx

```diff
@@ -532,6 +538,23 @@ const PlansFeaturesMain = ( {
 			  )
 			: undefined;
 
+	/**
+	 * Ensure the spotlight plan shows the price with which the plans was purchased.
+	 */
+	if (
+		gridPlanForSpotlight &&
+		purchase &&
+		gridPlanForSpotlight.planSlug === purchase.productSlug
+	) {
+		if ( gridPlanForSpotlight.isMonthlyPlan ) {
+			gridPlanForSpotlight.pricing.originalPrice.monthly = purchase.priceInteger;
+			gridPlanForSpotlight.pricing.originalPrice.full = Math.floor( purchase.priceInteger * 12 );
+		} else {
+			gridPlanForSpotlight.pricing.originalPrice.monthly = Math.floor( purchase.priceInteger / 12 );
+			gridPlanForSpotlight.pricing.originalPrice.full = purchase.priceInteger;
+		}
+	}
```

Comment:
I don't think we should be modifying these objects directly inline here. There's a `usePricingMetaForGridPlans` hook that's responsible for putting together pricing for the grid plans. You can use a Calypso state selector there 

> You can use a Calypso state selector there

Unless there's a respective selector from `@automattic/data-stores`, which we'd prefer to use instead. Anything else is just adding to the milestone of migrating pricing to NPM.

Reply:
Thanks. It looks great. Just needs some testing. Also not sure about https://github.com/Automattic/wp-calypso/pull/82761#discussion_r1350269902

> Just needs some testing

I need to run away in a bit. code-wise this looks good to me. I can help with more testing later or tomorrow